### PR TITLE
fix(v2): treat inline code in raw HTML as native element

### PR DESCRIPTION
**Pull request by lex111**
## Motivation

One of our users complained that inline code in raw HTML was not rendered correctly. I advised him to use Markdown syntax to define tables, to which he replied that in this case lists cannot be used. That's fair, raw HTML gives you more opportunities and flexibility,  so we should take this into account.

I mean, inline code in raw HTML should be rendered using regular HTML element, rather `CodeBlock` component (see screenshots below).

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

```
<table>
  <tr>
    <th>Argument</th>
  </tr>
  <tr>
    <td><code>name</code></td>
  </tr>
</table>
```

| Before   | After    |
| -------- | -------- |
| Screenshot from 2020-06-01 13-35-52 | image |

In other words, after that inline code in tables defined in different ways (Markdown and HTML) are rendered identically.

```
| Argument |
| -------- |
| `name`   |
```

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: packages/docusaurus-theme-classic/src/theme/MDXComponents/index.js [hunk 1]
```diff
@@ -16,6 +16,9 @@ export default {
   code: (props) => {
```
Comment (yangshun) [inline packages/docusaurus-theme-classic/src/theme/MDXComponents/index.js:16]: I wonder why we even map code to CodeBlock in the first place. Endi did this in the first commit of this file. Any idea?

File: packages/docusaurus-theme-classic/src/theme/MDXComponents/index.js [hunk 2]
```diff
@@ -16,6 +16,9 @@ export default {
...
     const {children} = props;
     if (typeof children === 'string') {
+      if (children.indexOf('\n') === -1) {
```
Comment (yangshun) [inline packages/docusaurus-theme-classic/src/theme/MDXComponents/index.js:19]: This makes sense. Suggestion:

```js
if (!children.includes('\n')) {
  ...
}
```

**Comment by yangshun** [comment]
@slorber mayb we can merge this PR, release a new version and then the typescript cheatsheet website won't have this issue.

File: packages/docusaurus-theme-classic/src/theme/MDXComponents/index.js (hunk 2 above)
Comment (lex111) [reply packages/docusaurus-theme-classic/src/theme/MDXComponents/index.js:19]: I did this because `indexOf` is the fastest of them.

File: packages/docusaurus-theme-classic/src/theme/MDXComponents/index.js (hunk 1 above)
Comment (lex111) [reply packages/docusaurus-theme-classic/src/theme/MDXComponents/index.js:16]: Nope :man_shrugging:

Comment (slorber) [reply packages/docusaurus-theme-classic/src/theme/MDXComponents/index.js:16]: This is what permits to transform ``` to the prism editor

**Comment by slorber** [comment]
Was working on the same thing at the same time ^^

I can confirm this solves the TS Cheatsheet use case. Also checked that single line code blocks get the editor and it seems to work too. 

image


Not a fan of checking for newline but if it works then it's fine ;)

File: packages/docusaurus-theme-classic/src/theme/MDXComponents/index.js (hunk 2 above)
Comment (lex111) [reply packages/docusaurus-theme-classic/src/theme/MDXComponents/index.js:19]: @yangshun so should I replace `indexOf` with `includes`? Or can I leave first one?

Comment (yangshun) [reply packages/docusaurus-theme-classic/src/theme/MDXComponents/index.js:19]: Yeah feel free. I think the perf difference here is negligible. I'd optimize for readability. But up to you.

Comment (lex111) [reply packages/docusaurus-theme-classic/src/theme/MDXComponents/index.js]: :heavy_check_mark:
